tests/instrumentation-framework/src/main/java/org/hawkular/apm/tests/dockerized/TestScenarioRunner.java

```diff
@@ -0,0 +1,230 @@
+     * @param apmServerPort - port at which Hawkular APM server will be started
+     * @throws DockerCertificateException
+     */
+    public TestScenarioRunner(String apmVersion, int apmServerPort) throws DockerCertificateException {
+        this.apmVersion = apmVersion;
+        this.apmServerPort = apmServerPort;
+
+        this.objectMapper = new ObjectMapper();
+        apmServer = new ApmMockServer();
+        apmServer.setPort(apmServerPort);
+        apmServer.setShutdownTimer(60*60*1000);
+    }
+
+    /**
+     * Run a test scenario
+     *
+     * @param testScenario
+     * @return number of successful test cases
+     */
+    public int run(TestScenario testScenario) {
+        if (testScenario.getEnvironment().getDockerCompose() != null &&
+                testScenario.getEnvironment().getImage() != null) {
+            throw new IllegalArgumentException("Ambiguous environment: defined docker image and docker-compose");
+        }
+
+        System.out.println("Starting test scenario: " + testScenario);
+        int successfulTestCases = 0;
+
+        for (TestCase test: testScenario.getTests()) {
+            if (test.isSkip()) {
+                continue;
+            }
+
+            TestEnvironmentExecutor testEnvironmentExecutor = testEnvironmentExecutor(testScenario);
+
+            try {
+                runTestCase(testScenario, test, testEnvironmentExecutor);
+                successfulTestCases++;
+            } catch (TestFailException ex) {
+                System.out.println("Test case failed: " + ex.toString());
+                System.out.println(ex.getMessage());
+                ex.printStackTrace();
+            }
+
+            testEnvironmentExecutor.close();
+        }
+
+        return successfulTestCases;
+    }
+
+    /**
+     * Run single test case
+     *
+     * @param testScenario Test scenario of the test case
+     * @param testCase Test case to run
+     * @param testEnvironmentExecutor Initialized environment
+     * @throws TestFailException
+     */
+    private void runTestCase(TestScenario testScenario, TestCase testCase,
+                             TestEnvironmentExecutor testEnvironmentExecutor) throws TestFailException {
+
+        System.out.println("Executing test case: " + testCase);
+
+        String environmentId = null;
+        try {
+            if (testEnvironmentExecutor instanceof DockerComposeExecutor) {
+                ((DockerComposeExecutor) testEnvironmentExecutor).createNetwork();
+            }
+
+            /**
+             * Start APM server
+             */
+            apmServer.setHost("0.0.0.0");
+            apmServer.setTraces(new ArrayList<>());
+            apmServer.run();
+
+            /**
+             * Run a container and wait
+             */
+            environmentId = testEnvironmentExecutor.run(testScenario.getEnvironment());
+            Thread.sleep(testScenario.getEnvironment().getInitWaitSeconds()*1000);
```

Comment:
It might, I  guess only time will tell :) The problem with this approach is that it might cause the CI to be unstable. You could perhaps add this description to the code as comments. 

If there are problems on CI with this code, you could add the `start` logic to a FutureTask, returning a Future. Your main code would call the `#get(1, TimeUnit.MINUTES)`. Your `start` logic would pool the container's port 9080, for /status, and would return a new Object once it returns what you expect to see from /status. This way, you don't need to wait N seconds *all* the time, no matter if the server spent a third of that time to be ready.